lib/Parser/Parser.cpp

```diff
@@ -1122,6 +1131,74 @@ bool Parser::parseLine(std::string &ErrStr) {
                   }
                   if (!consumeToken(ErrStr))
                     return false;
+                } else if (CurTok.str() == "range") {
+                  // look for =
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::Eq) {
+                    ErrStr = makeErrStr(TP, "expected '=' for range as 'range='");
+                    return false;
+                  }
+
+                  // look for [ (inclusive paren)
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::OpenBracket) {
+                    ErrStr = makeErrStr(TP, "expected '[' to specify lower bound of range");
+                    return false;
+                  }
+
+                  // look for untypedconst
+                  if (!consumeToken(ErrStr))
+                    return false;
+                  if (CurTok.K != Token::UntypedInt && CurTok.K == Token::Int) {
```

Comment:
what purpose does this check serve?

Reply:
I wanted to handle this separately so that if someone writes Souper IR directly and mentions lower/upper bound as '5:i32' (which is Token::Int) should get an informative error message that bounds must be specified without width. I can simplify this error check to this actually:
if (CurTok.K == Token::Int) {...}